applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.c

```diff
@@ -193,76 +216,57 @@ static int adv_create(void)
-	if (IS_ENABLED(CONFIG_BT_AUDIO_USE_BROADCAST_NAME_ALT)) {
-		ext_ad[1] = (struct bt_data)BT_DATA(BT_DATA_BROADCAST_NAME,
-						    CONFIG_BT_AUDIO_BROADCAST_NAME_ALT,
-						    sizeof(CONFIG_BT_AUDIO_BROADCAST_NAME_ALT) - 1);
-	} else {
-		ext_ad[1] = (struct bt_data)BT_DATA(BT_DATA_BROADCAST_NAME,
-						    CONFIG_BT_AUDIO_BROADCAST_NAME,
-						    sizeof(CONFIG_BT_AUDIO_BROADCAST_NAME) - 1);
-	}
+	sys_put_le24(broadcast_id, &brdcst_id_buf[BROADCAST_ID_START]);
 
-	/* Setup extended advertising data */
-	net_buf_simple_add_le16(&brdcst_id_buf, BT_UUID_BROADCAST_AUDIO_VAL);
-	net_buf_simple_add_le24(&brdcst_id_buf, broadcast_id);
-
-	ext_ad[2].data_len = brdcst_id_buf.len;
-	ext_ad[2].type = BT_DATA_SVC_DATA16;
-	ext_ad[2].data = brdcst_id_buf.data;
+	ret = bt_mgmt_adv_buffer_put(adv_buf, adv_buf_vacant, &adv_buf_cnt, BT_DATA_SVC_DATA16,
+				     ARRAY_SIZE(brdcst_id_buf), &brdcst_id_buf[0]);
+	if (ret) {
+		return ret;
+	}
 
-	net_buf_simple_add_le16(&brdcst_appearance_buf, CONFIG_BT_DEVICE_APPEARANCE);
+	sys_put_le16(CONFIG_BT_DEVICE_APPEARANCE, &brdcst_appearance_buf[0]);
 
-	ext_ad[3].data_len = brdcst_appearance_buf.len;
-	ext_ad[3].type = BT_DATA_GAP_APPEARANCE;
-	ext_ad[3].data = brdcst_appearance_buf.data;
+	ret = bt_mgmt_adv_buffer_put(adv_buf, adv_buf_vacant, &adv_buf_cnt, BT_DATA_GAP_APPEARANCE,
+				     ARRAY_SIZE(brdcst_appearance_buf), &brdcst_appearance_buf[0]);
+	if (ret) {
+		return ret;
+	}
 
 #if (CONFIG_AURACAST)
-	uint8_t pba_features = 0;
-	public_broadcast_features_set(&pba_features);
-
-	net_buf_simple_add_le16(&pba_buf, 0x1856);
-	net_buf_simple_add_u8(&pba_buf, pba_features);
-
-	/* Metadata */
-	/* 3 bytes for parental_rating and 3 bytes for active_flag LTVs */
-	net_buf_simple_add_u8(&pba_buf, 0x06);
-
-	/* Parental rating*/
-	/* Length */
-	net_buf_simple_add_u8(&pba_buf, 0x02);
-	/* Type */
-	net_buf_simple_add_u8(&pba_buf, BT_AUDIO_METADATA_TYPE_PARENTAL_RATING);
-	/* Value */
-	net_buf_simple_add_u8(&pba_buf, CONFIG_BT_AUDIO_BROADCAST_PARENTAL_RATING);
-
-	/* Active flag */
-	/* Length */
-	net_buf_simple_add_u8(&pba_buf, 0x02);
-	/* Type */
-	net_buf_simple_add_u8(&pba_buf, BT_AUDIO_METADATA_TYPE_AUDIO_STATE);
-	/* Value */
-	net_buf_simple_add_u8(&pba_buf, BT_AUDIO_ACTIVE_STATE_ENABLED);
+	public_broadcast_features_set(&pba_buf[PBA_FEATURE_START]);
 
-	/* If any additional data is to be added, remember to increase NET_BUF size */
+	pba_buf[BROADCAST_PARENTAL_RATING_START] = BT_AUDIO_METADATA_TYPE_PARENTAL_RATING;
 
-	ext_ad[4].data_len = pba_buf.len;
-	ext_ad[4].type = BT_DATA_SVC_DATA16;
-	ext_ad[4].data = pba_buf.data;
+	/* If any additional data is to be added, remember to increase NET_BUF size */
+	ret = bt_mgmt_adv_buffer_put(adv_buf, adv_buf_vacant, &adv_buf_cnt, BT_DATA_SVC_DATA16,
+				     ARRAY_SIZE(pba_buf), &pba_buf[0]);
+	if (ret) {
+		return ret;
+	}
 #endif /* (CONFIG_AURACAST) */
 
+	return adv_buf_cnt;
+}
+
+int broadcast_source_per_populate(struct bt_data *per_buf, size_t per_buf_vacant)
```

Comment:
Discuss here. There is something I don't quite get with respect to:` per_buf_vacant` being sent in, `per_buf_cnt `being returned and `struct bt_data` having data_len in the struct.

Reply:
We need count for the bt_mgmt_adv_start(). The vacant counter is to be consistent with the extended advertising populating. It could be removed in both if you feel it makes it clearer?

Not sure what you mean about struct bt_data!?